src/decisionengine/framework/engine/de_client.py

```diff
@@ -6,9 +6,10 @@
 
 def create_parser():
     parser = argparse.ArgumentParser()
-    parser.add_argument("--port", metavar="<port number>", default="8888", help="Default port is 8888")
-    parser.add_argument("--host", metavar="<hostname>", default="localhost", help="Default hostname is 'localhost'")
-    parser.add_argument(
+    optional = parser.add_argument_group("optional arguments")
+    optional.add_argument("--port", metavar="<port number>", default="8888", help="Default port is 8888")
+    optional.add_argument("--host", metavar="<hostname>", default="localhost", help="Default hostname is 'localhost'")
+    optional.add_argument(
```

Comment:
Aren't arguments starting with "-" or "--" considered optional arguments by default? What's the advantage of making it explicit? Same in the other files as well. Is this because of the unit tests? They would not get the default?